routes/appendLocals.js

```diff
@@ -63,7 +63,7 @@ function generateSRI(file) {
 
 function appendLocals(req, res) {
     const siteUrl = getCurrentSiteurl(req);
-    const canonicalUrl = `${config.siteurl}${req.path}`;
+    const canonicalUrl = `${config.siteurl}${req.originalUrl}`;
```

Comment:
I'm pretty sure this worked up some point.

Reply:
OK, this broke after b0238932d9c0b9fa5c904cf990bdf29e7a814922

Inside say about.js:

```js
console.log(req.originalUrl);   // /about/
console.log(req.path);          // /
console.log(req.pathname);      // undefined
console.log(req.url);           // /
```

I'm not sure how to fix this and keep this modular structure...

Reply:
Is it working as expected with this change to `req.originalUrl`?

Reply:
Yes, but `originalUrl` contains the query params which isn't what we want. I verified that this happens after  the aforementioned commit. For example, [this](https://github.com/MaxCDN/bootstrapcdn/tree/develop-xmr-before-modularize-app) branch doesn't have the canonical URL issue.

So, something is overwritten somewhere or something like that.

Reply:
If we don't find a solution soon, I'll need to revert the whole modularization and I don't want to do it :/

But I don't know how to solve this issue. I could possibly work around it by using `url.parse`, but I feel we are doing something wrong in the Router.